app/form_models/copy_vacancy_form.rb

```diff
@@ -0,0 +1,23 @@
+class CopyVacancyForm < VacancyForm
+  include ActiveModel::Model
+
+  delegate :starts_on_yyyy, :starts_on_mm, :starts_on_dd,
+           :ends_on_dd, :ends_on_mm, :ends_on_yyyy,
+           :expires_on_dd, :expires_on_mm, :expires_on_yyyy,
+           :publish_on_dd, :publish_on_mm, :publish_on_yyyy,
+           :errors, to: :vacancy
+
+  def initialize(vacancy:)
```

Comment:
This is here as a named parameter so that if someone forgets to provide it to the initialiser, rails will error with a super helpful message saying it's missing a value, rather than an obscure nil or active record not found error as it tries to initialise. I haven't set a default value on the right which makes this a mandatory field.

They're also helpful as it removes the issue of order dependency from parameters passed to methods, which becomes clearer when there's more than one :)

Reply:
Oh sweet. That's v. handy